src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -745,4 +946,201 @@ private void createTwoWindingsTransformer(Network network, VoltageLevel voltageL
+                                                                         Object attributeValue,
+                                                                         ReporterModel reporter,
+                                                                         Reporter subReporter) {
+        Network network = listener.getNetwork();
+        UUID networkUuid = listener.getNetworkUuid();
+
+        return doAction(listener, () -> {
+            if (listener.isApplyModifications()) {
+                Identifiable<?> identifiable = network.getIdentifiable(equipmentId);
+                if (identifiable == null) {
+                    throw new NetworkModificationException(EQUIPMENT_NOT_FOUND, equipmentId);
+                }
+                if (identifiable instanceof Switch) {
+                    changeSwitchAttribute((Switch) identifiable, attributeName, attributeValue);
+                } else if (identifiable instanceof Injection) {
+                    if (identifiable instanceof Generator) {
+                        changeGeneratorAttribute((Generator) identifiable, attributeName, attributeValue);
+                    }
+                } else if (identifiable instanceof Branch) {
+                    if (identifiable instanceof Line) {
+                        changeLineAttribute((Line) identifiable, attributeName, attributeValue);
+                    } else if (identifiable instanceof TwoWindingsTransformer) {
+                        changeTwoWindingsTransformerAttribute((TwoWindingsTransformer) identifiable, attributeName, attributeValue);
+                    }
+                } else if (identifiable instanceof ThreeWindingsTransformer) {
+                    changeThreeWindingsTransformerAttribute((ThreeWindingsTransformer) identifiable, attributeName, attributeValue);
+                } else if (identifiable instanceof HvdcLine) {
+                    // no hvdc line modifications yet
+                }
+            }
+        }, MODIFICATION_ERROR, networkUuid, reporter, subReporter);
+    }
+
+    public List<EquipmenModificationInfos> applyModifications(Network network, UUID networkUuid, RealizationInfos realizationInfos) {
+        // Apply all modifications belonging to the modification groups uuids in realizationInfos
+        List<EquipmenModificationInfos> allModificationsInfos = new ArrayList<>();
+        NetworkStoreListener listener = NetworkStoreListener.create(network,
+            networkUuid,
+            realizationInfos.getDestinationVariantId(),
+            null,
+            modificationRepository,
+            equipmentInfosService,
+            true,
+            true);
+        ReporterModel reporter = new ReporterModel("Realizaton", "Realization");
+
+        modificationRepository.getModificationsEntities(realizationInfos.getModifications()).forEach(modificationEntity -> {
+            ModificationType type = ModificationType.valueOf(modificationEntity.getType());
+            switch (type) {
```

Comment:
Need to have an object oriented approach ?
Review the design with a `Modification` interface and a class for each type which implements the interface
and a factory to create these classes
```
interface Modification {
    List<EquipmenModificationInfos> apply();
    
    void store();
}
```
1. For a modification from the front we create the modification then apply it and store it
2. For a realization get the modification from the database then  apply it only

The listener must be used only for the application.